crates/router/src/connector/powertranz/transformers.rs

```diff
@@ -1,210 +1,336 @@
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
 pub struct PowertranzPaymentsRequest {
-    amount: i64,
-    card: PowertranzCard,
+    transaction_identifier: String,
+    total_amount: f64,
+    currency_code: String,
+    three_d_secure: bool,
+    source: Source,
+    order_identifier: String,
+    billing_address: Option<PowertranzAddressDetails>,
+    shipping_address: Option<PowertranzAddressDetails>,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum Source {
+    Card(PowertranzCard),
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
 pub struct PowertranzCard {
-    name: Secret<String>,
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+    cardholder_name: Secret<String>,
+    card_pan: cards::CardNumber,
+    card_expiration: Secret<String>,
+    card_cvv: Secret<String>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct PowertranzAddressDetails {
+    first_name: Option<Secret<String>>,
+    last_name: Option<Secret<String>>,
+    line1: Option<Secret<String>>,
+    line2: Option<Secret<String>>,
+    city: Option<String>,
+    country: Option<enums::CountryAlpha2>,
+    state: Option<Secret<String>>,
+    postal_code: Option<Secret<String>>,
+    email_address: Option<Email>,
+    phone_number: Option<Secret<String>>,
 }
 
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for PowertranzPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        match item.request.payment_method_data.clone() {
-            api::PaymentMethodData::Card(req_card) => {
-                let card = PowertranzCard {
-                    name: req_card.card_holder_name,
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.request.amount,
-                    card,
+        let source = match item.request.payment_method_data.clone() {
+            api::PaymentMethodData::Card(card) => Ok(Source::from(&card)),
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Payment method".to_string(),
+            )),
+        }?;
+        let billing_address = get_address_details(&item.address.billing, &item.request.email);
+        let shipping_address = get_address_details(&item.address.shipping, &item.request.email);
+        Ok(Self {
+            transaction_identifier: Uuid::new_v4().to_string(),
```

Comment:
Shouldn't we be sending an identifier that we store as the transaction identifier? Wouldn't that help us with recon later, if required?

Reply:
transaction_identifier should be in GUID format, we are passing our payment id in order_identifier field

Reply:
Okay I figured out that we're storing it as connector transaction ID. Should be fine, marking this resolved.